### PR TITLE
New rogue tables

### LINKED ISSUE #79: Build out first round of new tables for Rogue for the FUTURE
_FEATURE OVERVIEW_
-------------------
#### User Story

-Tables: sign up, photo, event, reviews

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Creates `events`, `signups`, `photos`, and `reviews` tables/models in rogue

#### How should this be reviewed?
Run migrations, get new tables? great.

#### Any background context you want to provide?

[Tech spec](https://docs.google.com/document/d/1JMlUHbXMe3Iyeo-tl3oXUJ98J5kWpeNew8JFI9f_Wdg/edit)

#### Relevant tickets
Fixes #79

### CODE REVIEW

File: database/migrations/2016_12_01_210658_create_events_table.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        Schema::create('events', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('northstar_id')->index()->comment('The users Northstar ID');
+            $table->string('event_type')->index()->comment('the type of event');
+            $table->string('submission_type')->index()->comment('Whether the event was User event or an Admin event');
+            $table->timestamps();
+        });
+    }
+
```
Comment (chloealee) [inline database/migrations/2016_12_01_210658_create_events_table.php]: do we need a `created_at` column?

Comment (chloealee) [inline database/migrations/2016_12_01_210658_create_events_table.php]: sorry nevermind - see that is there with timestamps!

**Comment by chloealee** [comment]
awesome!! 👍

File: database/migrations/2016_12_02_153413_create_reviews_table.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+            $table->integer('signup_id')->unsigned()->comment('The signup this Post references.');
+            $table->foreign('signup_id')->references('id')->on('signups');
+
+            $table->string('northstar_id')->index()->comment('The user that submitted the thing being reviewed
+');
+            $table->string('admin_northstar_id')->comment('The NS id of the admin doing the reviewing');
+
+            $table->string('status')->nullable();
+            $table->string('old_status')->nullable();
```
Comment (angaither) [inline database/migrations/2016_12_02_153413_create_reviews_table.php]: what's this `old_status` for? will it contain every old status if multiple people review? or just the previous one?

Comment (sbsmith86) [inline database/migrations/2016_12_02_153413_create_reviews_table.php]: I _think_ the idea was just the previous one. so we have a log of a history of changes with this combined with the `events` table.

File: database/migrations/2016_12_01_210658_create_events_table.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('events', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('northstar_id')->index()->comment('The users Northstar ID');
+            $table->string('event_type')->index()->comment('the type of event');
```
Comment (angaither) [inline database/migrations/2016_12_01_210658_create_events_table.php]: i think `northstar_id` and `event_type` should probably have indexes

Comment (sbsmith86) [inline database/migrations/2016_12_01_210658_create_events_table.php]: @angaither wait `event_type` does have an index on it. are you saying to remove it? or keep it?

**Comment by sbsmith86** [comment]
@angaither I added more indexes to the tables mostly on the `event_id` and `signup_id` fields. Let me know if I missed any.

**Comment by angaither** [comment]
@sbsmith86 looks 🆗  the only thing I'd say might be `status` so we can easily query for all accepted RBs

**Comment by sbsmith86** [comment]
Ah yes. good idea.

**Comment by sbsmith86** [comment]
@angaither Updated!

File: database/migrations/2016_12_01_222513_create_signups_table.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    public function up()
+    {
+        Schema::create('signups', function (Blueprint $table) {
+            $table->increments('id')->comment('Primary key.');
+            $table->integer('event_id')->unsigned()->comment('The event ID');
+            $table->foreign('event_id')->references('id')->on('events');
+            $table->string('northstar_id')->index()->comment('The users Northstar ID');
+            $table->integer('campaign_id')->index()->comment('The campaign node id the event happened on');
+            $table->integer('campaign_run_id')->index()->comment('The campaign run node id the event happened on.');
```
Comment (sbsmith86) [inline database/migrations/2016_12_01_222513_create_signups_table.php]: @angaither hmm? they are indexes?

**Pull request merged**